testmark.go

```diff
@@ -74,3 +80,25 @@ type DirEnt struct {
 	Children     map[string]*DirEnt
 	ChildrenList []DirEnt
 }
+
+func (doc *Document) describe(ent *DirEnt) []string {
+	result := []string{}
+	if ent == nil {
+		return result
+	}
+	if ent.Hunk != nil {
+		loc := doc.HunksByName[ent.Hunk.Name].LineStart + 1
+		s := fmt.Sprintf("%s:%d:%s", doc.Path, loc, ent.Hunk.Name)
+		result = append(result, s)
+	}
+	for _, child := range ent.ChildrenList {
+		result = append(result, doc.describe(&child)...)
+	}
+	return result
+}
+
+func (doc *Document) Describe(ent *DirEnt) []string {
```

Comment:
It might be sensible to make `Describe` a method of `DirEnt` and take the document as a parameter.